src/NServiceBus.Transport.Msmq/DelayedDelivery/DueDelayedMessagePoller.cs

```diff
@@ -0,0 +1,341 @@
+
+        async Task WaitForSignal(CancellationToken cancellationToken)
+        {
+            try
+            {
+                await signalQueue.Reader.ReadAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
+            {
+                //Shutting down
+            }
+        }
+
+        Task WaitIfNeeded(DateTimeOffset nextPoll, CancellationToken cancellationToken)
+        {
+            var waitTime = nextPoll - DateTimeOffset.UtcNow;
+            if (waitTime > TimeSpan.Zero)
+            {
+                return Task.Delay(waitTime, cancellationToken);
+            }
+
+            return Task.CompletedTask;
+        }
+
+        async Task Poll(TaskCompletionSource<DateTimeOffset?> result, CancellationToken cancellationToken)
+        {
+            DelayedMessage timeout = null;
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+            try
+            {
+                using (var tx = new TransactionScope(TransactionScopeOption.Required, transactionOptions, TransactionScopeAsyncFlowOption.Enabled))
+                {
+                    timeout = await delayedMessageStore.FetchNextDueTimeout(now, cancellationToken).ConfigureAwait(false);
+                    fetchCircuitBreaker.Success();
+
+                    if (timeout != null)
+                    {
+                        result.SetResult(null);
+                        await HandleDueDelayedMessage(timeout, cancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        using (new TransactionScope(TransactionScopeOption.RequiresNew, transactionOptions, TransactionScopeAsyncFlowOption.Enabled))
+                        {
+                            var nextDueTimeout = await delayedMessageStore.Next(cancellationToken).ConfigureAwait(false);
+                            if (nextDueTimeout.HasValue)
+                            {
+                                result.SetResult(nextDueTimeout);
+                            }
+                            else
+                            {
+                                //If no timeouts, wait a while
+                                result.SetResult(now.Add(MaxSleepDuration));
+                            }
+                        }
+                    }
+
+                    tx.Complete();
+                }
+            }
+            catch (QueueNotFoundException exception)
+            {
+                if (timeout != null)
+                {
+                    await TrySendDelayedMessageToErrorQueue(timeout, exception, cancellationToken).ConfigureAwait(false);
+                }
+            }
+            catch (OperationCanceledException) when (cancellationToken.IsCancellationRequested)
+            {
+                //Shutting down
+                throw;
+            }
+            catch (Exception exception)
```

Comment:
OR 

Not invoking `catch (Exception exception)` when cancelled.
```suggestion
            catch (Exception exception) when (!cancellationToken.IsCancellationRequested)
```

Reply:
I'm on the fence here. If we return the OperationCanceledException is swallowed, could it be that it needs to bubble up?

Reply:
@lailabougria No, it would not be swallowed but just thrown as it isn't matching the generic Exception catch.